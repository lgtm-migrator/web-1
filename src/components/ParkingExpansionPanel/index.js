import ExpansionPanel from "@material-ui/core/ExpansionPanel"
import ExpansionPanelSummary from "@material-ui/core/ExpansionPanelSummary"
import Typography from "@material-ui/core/Typography"
import ExpansionPanelDetails from "@material-ui/core/ExpansionPanelDetails"
import GridList from "@material-ui/core/GridList"
import GridListTile from "@material-ui/core/GridListTile"
import EmbedMap from "../../containers/EmbedGoogleMap"
import GridListTileBar from "@material-ui/core/GridListTileBar"
import React from "react"
import Link from "@material-ui/core/Link"
import { Image } from "muy"

export default ({
  car_front_photo_height,
  car_front_photo_uri,
  car_front_photo_width,
  car_plate,
  coordinates,
  created_at,
  uuid,
}) => (
  <ExpansionPanel id={`parking-${uuid}`} key={uuid}>
    <ExpansionPanelSummary>
      <header>
        <Typography>{car_plate}</Typography>
        <Typography>Data e hora da denÃºncia: {created_at}</Typography>
        <Link>Expandir para mais detalhes</Link>
      </header>
    </ExpansionPanelSummary>
    <ExpansionPanelDetails>
      <GridList cellHeight={289} cols={3}>
        {coordinates && (
          <GridListTile cols={3}>
            <EmbedMap
              coordinates={`${coordinates.x},${coordinates.y}`}
              height={450}
              variant="cover"
              width={600}
            />
          </GridListTile>
        )}
        <GridListTile cols={3}>
          <Image
            height={car_front_photo_height}
            loading="lazy"
            src={car_front_photo_uri}
            variant="cover"
            width={car_front_photo_width}
          />
          <GridListTileBar title="Foto da frente" />
        </GridListTile>
      </GridList>
    </ExpansionPanelDetails>
  </ExpansionPanel>
)
