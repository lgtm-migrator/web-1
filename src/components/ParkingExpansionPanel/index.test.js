import React from "react"
import renderer from "react-test-renderer"
import { render } from "@testing-library/react"
import ParkingExpansionPanel from "."

describe("ParkingExpansionPanel", () => {
  const car_front_photo_height = "car_front_photo_height"
  const car_front_photo_uri = "car_front_photo_uri"
  const car_front_photo_width = "car_front_photo_width"
  const car_plate = "car_plate"
  const car_rear_photo_height = "car_rear_photo_height"
  const car_rear_photo_uri = "car_rear_photo_uri"
  const car_rear_photo_width = "car_rear_photo_width"
  const coordinates = { x: 0, y: 1 }
  const created_at = "created_at"
  const uuid = "uuid"

  const props = {
    car_front_photo_height,
    car_front_photo_uri,
    car_front_photo_width,
    car_plate,
    car_rear_photo_height,
    car_rear_photo_uri,
    car_rear_photo_width,
    coordinates,
    created_at,
    uuid,
  }

  it("should render correctly", () => {
    const tree = renderer.create(<ParkingExpansionPanel {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("should not render the map when coordinates are not defined", () => {
    props.coordinates = null

    const { container } = render(<ParkingExpansionPanel {...props} />)
    const iframe = container.querySelector("iframe")

    expect(iframe).toBeFalsy()
  })
})
