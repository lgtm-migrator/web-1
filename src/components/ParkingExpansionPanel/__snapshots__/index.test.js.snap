// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParkingExpansionPanel should render correctly 1`] = `
<div
  className="MuiPaper-root MuiExpansionPanel-root MuiExpansionPanel-rounded MuiPaper-elevation1 MuiPaper-rounded"
  id="parking-uuid"
>
  <div
    aria-disabled={false}
    aria-expanded={false}
    className="MuiButtonBase-root MuiExpansionPanelSummary-root"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiExpansionPanelSummary-content"
    >
      <header>
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          car_plate
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          Data e hora da den√∫ncia: 
          created_at
        </p>
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          Expandir para mais detalhes
        </a>
      </header>
    </div>
  </div>
  <div
    className="MuiCollapse-container MuiCollapse-hidden"
    style={
      Object {
        "minHeight": "0px",
      }
    }
  >
    <div
      className="MuiCollapse-wrapper"
    >
      <div
        className="MuiCollapse-wrapperInner"
      >
        <div
          role="region"
        >
          <div
            className="MuiExpansionPanelDetails-root"
          >
            <ul
              className="MuiGridList-root"
              style={
                Object {
                  "margin": -2,
                }
              }
            >
              <li
                className="MuiGridListTile-root"
                style={
                  Object {
                    "height": 293,
                    "padding": 2,
                    "width": "100%",
                  }
                }
              >
                <div
                  className="MuiGridListTile-tile"
                >
                  <iframe
                    allowFullScreen={true}
                    className="MuiBox-root MuiBox-root-91 makeStyles-cover-89"
                    frameBorder={0}
                    src="https://www.google.com/maps/embed/v1/place?q=0,1&key=undefined"
                  />
                </div>
              </li>
              <li
                className="MuiGridListTile-root"
                style={
                  Object {
                    "height": 293,
                    "padding": 2,
                    "width": "100%",
                  }
                }
              >
                <div
                  className="MuiGridListTile-tile"
                >
                  <img
                    className="makeStyles-root-92 makeStyles-cover-93"
                    height="car_front_photo_height"
                    loading="lazy"
                    src="car_front_photo_uri"
                    width="car_front_photo_width"
                  />
                  <div
                    className="MuiGridListTileBar-root MuiGridListTileBar-titlePositionBottom"
                  >
                    <div
                      className="MuiGridListTileBar-titleWrap"
                    >
                      <div
                        className="MuiGridListTileBar-title"
                      >
                        Foto da frente
                      </div>
                    </div>
                  </div>
                </div>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
